File: uri/structures/1069.cpp
// Diamonts and Sand
// Balaceamento de parenteses. Pilha

#include <iostream>
#include <stack>

using namespace std;

int main(){
	unsigned int N;
	cin >> N;

	while(N--){
		int cont = 0;
		string mine;
		stack<char> diamonts;
		
		cin >> mine;

		for(auto i: mine){
			if(i == '<'){
				diamonts.push(i);
			}
			else if(i == '>'){
				if(!diamonts.empty()){
					diamonts.pop();
					cont++;
				}			
			}
		}

		cout << cont << endl;
	}
	return 0;
}

File: uri/structures/1069.py
# Diamonts and Sand

def main():
	N = int(raw_input())

	while(N):
		cont = 0;
		mine = ""
		diamonts = []
		
		mine = raw_input()

		for i in mine:
			if(i == '<'):
				diamonts.append(i)
		
			elif(i == '>'):
				if(diamonts):
					diamonts.pop();
					cont += 1;	

		print cont
		N = N - 1

if __name__ == '__main__':
	main()
